fhirstore/search/searcharguments.py

```diff
@@ -0,0 +1,172 @@
+from collections import defaultdict
+from collections.abc import Mapping
+from werkzeug.datastructures import ImmutableMultiDict
+
+import re
+
+
+def url_to_dict(url_args):
+    if isinstance(url_args, ImmutableMultiDict):
+        search_args = {key: url_args.getlist(key) for key in url_args.keys()}
+        return search_args
+    else :
+        return url_args
+ 
+    
+def parse_comma(key, value):
+    if "," in value:
+        return "multiple", {key: value.split(",")}
+    else:
+        return key, [value]
+
+
+def pre_process_params(url_args):
+    search_args = url_to_dict(url_args)
+    processed_params = defaultdict(list)
+    if search_args == {}:
+        processed_params = {}
+    for key, value in search_args.items():
+        if len(value) == 1:
+            parsed_key, parsed_value = parse_comma(key, value[0])
+            processed_params.update({parsed_key: parsed_value})
+        else:
+            for element in value:
+                parsed_key, parsed_value = parse_comma(key, element)
+                if parsed_key == "multiple":
+                    processed_params[parsed_key] = parsed_value
+                else:
+                    processed_params[parsed_key].append(parsed_value[0])
+    return processed_params
+
+class ReverseChain:
```

Comment:
Could you add some doc for this new feature?

Reply:
done